src/tools/pbs_sleep.c

```diff
@@ -37,33 +37,47 @@
  */
 /**
  * @file
- *		pbs_sleep_win.c
+ *		pbs_sleep.c
  *
  * @brief
- *		This file contains functions related to sleep of PBS on Windows.
+ *		This file contains functions related to sleep of PBS.
  *
  * Functions included are:
  * 	main()
  *
  */
 #include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#ifdef WIN32
 #include <windows.h>
 #include "win.h"
+#endif
+
 /**
  * @Brief
- *      This is main function of pbs_sleep_win process.
- *      It calls sleep internally for the number of seconds passed to it.
+ *      This is main function of pbs_sleep process.
+ *      It calls sleep internally for the number of seconds passed to it, -1 for sleep indefinitely.
  *
  */
+
+int
 main(int argc, char *argv[])
 {
 	if (argc != 2) {
 		fprintf(stderr, "%s secs\n", argv[0]);
 		exit(1);
 	}
 
+	/* if argv[1] is -1, loop with sleep 1 indefinitely */
+	if (strcmp(argv[1], "-1")) {
+		while(1)
+			sleep(1);
+	}
+
 	sleep(atoi(argv[1]));
```

Comment:
Instead of doing one sleep(), do a loop around sleep(1) like above.  The sleep() function doesn't work well with a suspended process.  The sleep() function will sleep for N seconds, regardless if the process is suspended or not.  If the process is suspended and the sleep time is in the past, the sleep() will end when the process is resumed.  If you loop around sleep(1), one of the sleep(1) calls will end and we'll immediately start another.